tests/Unit/inc/classes/ImagifyUser/TestGetPercentConsumedQuota.php

```diff
@@ -0,0 +1,103 @@
+<?php
+namespace Imagify\tests\Unit\inc\classes\ImagifyUser;
+
+use Brain\Monkey\Functions;
+use Imagify\tests\Unit\TestCase;
+use Mockery;
+
+use Imagify_Data;
+use Imagify_User;
+
+/**
+ * Tests for Imagify_User->get_percent_consumed_quota().
+ *
+ * @covers Imagify_User::get_percent_consumed_quota
+ * @group  ImagifyAPI
+ */
+class Test_GetPercentConsumedQuota extends TestCase {
+
+	/**
+	 * Test Imagify_User->get_percent_consumed_quota() should return 0 when couldn’t fetch user account data.
+	 */
+	public function testShouldReturnZeroWhenCouldNotFetchUserData() {
+		$wp_error = $this->getMockBuilder( '\WP_Error' )
+			->disableOriginalConstructor()
+			->setConstructorArgs( [ 'error_id', 'Error Message' ] )
+			->getMock();
+
+		Functions\when( 'get_imagify_user' )->justReturn( $wp_error );
+		Functions\expect( 'imagify_round_half_five' )->never();
+
+		$this->assertSame( ( new Imagify_User() )->get_percent_consumed_quota(), 0 );
+	}
+
+	/**
+	 * Test Imagify_User->get_percent_consumed_quota() should return a quota when able to fetch user account data.
+	 */
+	public function testShouldReturnQuotaWhenFetchedUserData() {
+		$userData = (object) [
+			'id'                           => 1,
+			'email'                        => 'imagify@example.com',
+			'plan_id'                      => '1',
+			'plan_label'                   => 'free',
+			'quota'                        => 1000,
+			'extra_quota'                  => 0,
+			'extra_quota_consumed'         => 0,
+			'consumed_current_month_quota' => 900, // Current consumed quota 90%.
+			'next_date_update'             => '',
+			'is_active'                    => 1,
+		];
+
+		Functions\when( 'get_imagify_user' )->justReturn( $userData );
+		Functions\when( 'imagify_round_half_five' )->alias(
+			function( $number ) {
+				$number = strval( $number );
+				$number = explode( '.', $number );
+
+				if ( ! isset( $number[1] ) ) {
+					return $number[0];
+				}
+
+				$decimal = floatval( '0.' . substr( $number[1], 0, 2 ) ); // Cut only 2 numbers.
+
+				if ( $decimal > 0 ) {
+					if ( $decimal <= 0.5 ) {
+						return floatval( $number[0] ) + 0.5;
+					}
+					if ( $decimal <= 0.99 ) {
+						return floatval( $number[0] ) + 1;
+					}
+					return 1;
+				}
+
+				return floatval( $number );
+			}
+		);
+
+		$imagify_data_mock = Mockery::mock( Imagify_Data::class );
+		// Change the Imagify_Data::$_instance to the mock.
+		$ref = $this->get_reflective_property( '_instance', Imagify_Data::class );
+
+		$ref->setValue( $imagify_data_mock, $imagify_data_mock );
```

Comment:
This code could be moved to the trait for reuse throughout the tests.

```php
$this->set_singleton_instance( Imagify_Data::class, $imagify_data_mock );
```

And the method on the Trait:

```php

/**
 * Set the singleton's `$_instance` property to the given instance.
 *
 * @param string $class_name Name of the target class.
 * @param mixed  $instance   Instance of the target object.
 *
 * @throws \ReflectionException Throws an exception if property does not exist.
 */
protected function set_singleton_instance( $class, $instance ) {
	$ref = $this->get_reflective_property( '_instance', $class );
	$ref->setValue( $instance, $instance );
}
```